google_guest_agent/windows_accounts.go

```diff
@@ -41,13 +41,20 @@ var (
 // requirements: https://technet.microsoft.com/en-us/library/cc786468.
 // Characters that are difficult for users to type on a command line (quotes,
 // non english characters) are not used.
-func newPwd() (string, error) {
+func newPwd(userPwLgth int) (string, error) {
 	pwLgth := 15
+	maxPwLgth := 255
 	lower := []byte("abcdefghijklmnopqrstuvwxyz")
 	upper := []byte("ABCDEFGHIJKLMNOPQRSTUVWXYZ")
 	numbers := []byte("0123456789")
 	special := []byte(`~!@#$%^&*_-+=|\(){}[]:;<>,.?/`)
 	chars := bytes.Join([][]byte{lower, upper, numbers, special}, nil)
+	if userPwLgth > maxPwLgth {
+		pwLgth = maxPwLgth
+	}
+	if userPwLgth > pwLgth {
```

Comment:
This logic is setting pwLgth to userPwLgth no matter if its greater than maxPwLgth.

Reply:
Great catch. I moved the user defined password length above the max length evaluation.